internal/services/codesigning/trusted_signing_certificate_profile_resource.go

```diff
@@ -0,0 +1,478 @@
+			}
+
+			if metadata.ResourceData.HasChange("profile_type") {
+				properties.Properties.ProfileType = model.ProfileType
+			}
+
+			if err := client.CreateThenPoll(ctx, *id, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r TrustedSigningCertificateProfileResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.CodeSigning.Client.CertificateProfiles
+
+			id, err := certificateprofiles.ParseCertificateProfileID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := TrustedSigningCertificateProfileModel{
+				Name:                    id.CertificateProfileName,
+				TrustedSigningAccountId: codesigningaccounts.NewCodeSigningAccountID(id.SubscriptionId, id.ResourceGroupName, id.CodeSigningAccountName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if properties := model.Properties; properties != nil {
+					state.Certificates = flattenCertificateModelArray(properties.Certificates)
+					state.IdentityValidationId = properties.IdentityValidationId
+					state.IncludeCity = pointer.From(properties.IncludeCity)
+					state.IncludeCountry = pointer.From(properties.IncludeCountry)
+					state.IncludePostalCode = pointer.From(properties.IncludePostalCode)
+					state.IncludeState = pointer.From(properties.IncludeState)
+					state.IncludeStreetAddress = pointer.From(properties.IncludeStreetAddress)
+					state.ProfileType = properties.ProfileType
+					state.Status = pointer.From(properties.Status)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r TrustedSigningCertificateProfileResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.CodeSigning.Client.CertificateProfiles
+
+			id, err := certificateprofiles.ParseCertificateProfileID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func flattenCertificateModelArray(inputList *[]certificateprofiles.Certificate) []CertificateModel {
+	outputList := make([]CertificateModel, 0)
+	if inputList == nil {
+		return outputList
+	}
+	for _, input := range *inputList {
+		output := CertificateModel{
+			Revocation: flattenRevocationModel(input.Revocation),
+		}
+
+		if input.CreatedDate != nil {
+			output.CreatedDate = *input.CreatedDate
+		}
```

Comment:
Maybe we can just use `pointer.From` to simplify this, and inline define all the below properties in the composite literal?

```suggestion
		output.CreatedDate = pointer.From(input.CreatedDate)
```